Models/IonAuthModel.php

```diff
@@ -0,0 +1,2798 @@
+<?php namespace IonAuth\Models;
+
+/**
+ * Name:    Ion Auth Model
+ * Author:  Ben Edmunds
+ *           ben.edmunds@gmail.com
+ * @benedmunds
+ *
+ * Added Awesomeness: Phil Sturgeon
+ *
+ * Created:  10.01.2009
+ *
+ * Description:  Modified auth system based on redux_auth with extensive customization. This is basically what Redux Auth 2 should be.
+ * Original Author name has been kept but that does not mean that the method has not been modified.
+ *
+ * Requirements: PHP 7.1 or above
+ *
+ * @package    CodeIgniter-Ion-Auth
+ * @author     Ben Edmunds
+ * @license    https://opensource.org/licenses/MIT	MIT License
+ * @link       http://github.com/benedmunds/CodeIgniter-Ion-Auth
+ * @filesource
+ */
+
+use \CodeIgniter\Database\ConnectionInterface;
+
+/**
+ * Class IonAuthModel
+ *
+ * @property Ion_auth $ion_auth The Ion_auth library
+ */
+class IonAuthModel
+{
+	/**
+	 * Max cookie lifetime constant
+	 */
+	const MAX_COOKIE_LIFETIME = 63072000; // 2 years = 60*60*24*365*2 = 63072000 seconds;
+
+	/**
+	 * Max password size constant
+	 */
+	const MAX_PASSWORD_SIZE_BYTES = 4096;
+
+	/**
+	 * @var Config\IonAuth
+	 */
+	private $config;
+
+	/**
+	 * @var \CodeIgniter\Session\Session
+	 */
+	private $session;
+
+	/**
+	 * Holds an array of tables used
+	 *
+	 * @var array
+	 */
+	public $tables = [];
+
+	/**
+	 * Activation code
+	 *
+	 * Set by deactivate() function
+	 * Also set on register() function, if email_activation
+	 * option is activated
+	 *
+	 * This is the value devs should give to the user
+	 * (in an email, usually)
+	 *
+	 * It contains the *user* version of the activation code
+	 * It's a value of the form "selector.validator"
+	 *
+	 * This is not the same activation_code as the one in DB.
+	 * The DB contains a *hashed* version of the validator
+	 * and a selector in another column.
+	 *
+	 * THe selector is not private, and only used to lookup
+	 * the validator.
+	 *
+	 * The validator is private, and to be only known by the user
+	 * So in case of DB leak, nothing could be actually used.
+	 *
+	 * @var string
+	 */
+	public $activation_code;
+
+	/**
+	 * Identity
+	 *
+	 * @var string
+	 */
+	public $identity;
+
+	/**
+	 * Where
+	 *
+	 * @var array
+	 */
+	public $_ion_where = [];
+
+	/**
+	 * Select
+	 *
+	 * @var array
+	 */
+	public $_ion_select = [];
+
+	/**
+	 * Like
+	 *
+	 * @var array
+	 */
+	public $_ion_like = [];
+
+	/**
+	 * Limit
+	 *
+	 * @var string
+	 */
+	public $_ion_limit = NULL;
+
+	/**
+	 * Offset
+	 *
+	 * @var string
+	 */
+	public $_ion_offset = NULL;
+
+	/**
+	 * Order By
+	 *
+	 * @var string
+	 */
+	public $_ion_order_by = NULL;
+
+	/**
+	 * Order
+	 *
+	 * @var string
+	 */
+	public $_ion_order = NULL;
```

Comment:
Set proper visibility and camelCase
```suggestion
	protected $ionOrder = NULL;
```